tests/src/test/java/alluxio/master/lineage/LineageMasterIntegrationTest.java

```diff
@@ -150,21 +152,25 @@ public void lineageRecoveryTest() throws Exception {
       out.write("foo".getBytes());
       out.close();
       lineageClient.reportLostFile("/testFile");
-      // Wait for the log file to be created by the recompute job
-      while (!logFile.exists()) {
-        CommonUtils.sleepMs(20);
-      }
-      // Wait for the output to be written (this should be very fast)
-      CommonUtils.sleepMs(10);
-      BufferedReader reader = new BufferedReader(new FileReader(logFile));
-      try {
-        Assert.assertEquals("hello world", reader.readLine());
-      } finally {
-        reader.close();
-      }
     } finally {
       lineageClient.close();
     }
+
+    // Wait for the log file to be created by the recompute job
+    CommonTestUtils.waitFor("the log file to be written", new Function<Void, Boolean>() {
+      @Override
+      public Boolean apply(Void input) {
+        if (!logFile.exists()) {
+          return false;
+        }
+        try (BufferedReader reader = new BufferedReader(new FileReader(logFile))) {
```

Comment:
Is the behavior different from before? Before it waited until the file exists, then tried to read it. The new version also just waits until the file exists and reads it. What allows the file to be flushed before reading it?